runner/app/live/pipelines/comfyui.py

```diff
@@ -0,0 +1,63 @@
+from comfystream.client import ComfyStreamClient
+
+import os
+import json
+import torch
+import torch.nn as nn
+from torchvision import transforms
+from PIL import Image
+import asyncio
+
+from .interface import Pipeline
+
+
+class ComfyUI(Pipeline):
+  def __init__(self, **params):
+    super().__init__(**params)
+    # configure as is needed
+    comfy_ui_workspace = os.getenv("COMFY_UI_WORKSPACE")
+    default_workflow = os.getenv("COMFY_UI_DEFAULT_WORKFLOW")
+
+    self.client = ComfyStreamClient(cwd=comfy_ui_workspace)
+    with open(default_workflow, "r") as f:
+      prompt = json.load(f)
+    self.client.set_prompt(prompt)
+
+    # Comfy will cache nodes that only need to be run once (i.e. a node that loads model weights)
+    # We can run the prompt once before actual inputs come in to "warmup"
+    # input = torch.randn(1, 512, 512, 3)
+    # self.client.queue_prompt(input)
+
+    # self.update_params(**params)
+
+  def process_frame(self, image: Image.Image) -> Image.Image:
+    # transform = transforms.ToTensor()
+    transform = transforms.Compose([
+      transforms.Resize((512, 512)),  # Resize the image to 512x512
+      transforms.ToTensor(),          # Convert the image to a tensor
+      transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])  # Normalize the image
+    ])
+
+    # Apply the transformation to convert to torch.Tensor
```

Comment:
There is a simpler way to convert a PIL.Image into a BHWC (ComfyUI expects this [shape](https://docs.comfy.org/essentials/custom_node_tensors#what-is-a-tensor)) torch.Tensor.

```
# Convert PIL.Image to numpy
# Normalize by dividing by 255 to ensure the tensor values are between 0 and 1
image_np = np.array(image).astype(np.float32) / 255.0
# Convert from numpy to torch.Tensor
# Initially, the torch.Tensor will have shape HWC but we want BHWC
# unsqueeze(0) will add a batch dimension at the beginning of 1 which means we just have 1 image
image_tensor = torch.tensor(image_np).unsqueeze(0)
```

A few other examples of this type of conversion for reference:
- The example server converts [from numpy to torch.Tensor](https://github.com/yondonfu/comfystream/blob/d6791a8d38d3b6953a340021eeda610ebc7dc530/server/pipeline.py#L19)
- The diffusers codebase contains helper functions like [pil_to_numpy](https://github.com/huggingface/diffusers/blob/5b972fbd6a6c50cf1afdf1ba34c34d84fc67861c/src/diffusers/image_processor.py#L157) and [numpy_to_pt][(https://github.com/huggingface/diffusers/blob/5b972fbd6a6c50cf1afdf1ba34c34d84fc67861c/src/diffusers/image_processor.py#L177).

Reply:
Nice, it worked. I also had to convert the image to "RGB", but then it started to work. Thanks Yondon!